### PR TITLE
Adding  form

### PR DESCRIPTION
**Pull request by osamaashraf6**
## What does this PR do?

This PR is raised to implement the following features: 👍 :

> In this project, The Blog app will be a classic example of a blog website. I will create a fully functional website that will show the list of posts and empower readers to interact with them by adding comments and liking posts.🪄🎈

- [ ] Create a method called current_user in your ApplicationController to make current user data available to all controllers.
- [ ] It will return the first user from the database.

## Description of the task to be completed 👎 

- [ ] Run a program using the command line.👌
- [ ] Create forms to perform the following functions:
- [ ] Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- [ ] Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- [ ] Allow Users to add likes to Posts.
- [ ] Initialize my project with Git.💯

  
## How should it be tested

Clone the repository
```
git clone https://github.com/osamaashraf6/Blog-App
```
Navigate to the folder
```
cd Blog-App
```

Checkout the branch
```
git checkout adding__form
```

#### Thanks for reviewing my PR 😊
Don't forget to give a ⭐️ to support me 😃

### CODE REVIEW

File: app/views/posts/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    <%= form.submit "Add post" ,class: "btn" %>
+  <% end %>
+</section>
+<%# <%= form_for :post do |f| %> 
+  <%# <%= f.label :title %
+  <%# <%= f.text_field :title %>
+  <%# <%= f.label :body %> 
+  <%# <%= f.text_area :body %> 
+  <%# <%= f.submit "Create Post" %> 
+<%# <% end %>
```
Comment (anagudelogu) [inline app/views/posts/new.html.erb]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+#     if @new_like.save
+#       @new_like.update_like_counter
+#       redirect_to user_post_path(user_id: @post.author_id, id: @post.id)
+#     else
+#       redirect_to user_post_path(user_id: @post.author_id, id: @post.id),
+#                   alert: 'error'
+#     end
+#   end
+# end
```
Comment (anagudelogu) [inline app/controllers/likes_controller.rb]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  end
+end
+
+# def create
+#   @post = Post.find(params[:post_id])
+#   @new_comment = Comment.create(comment_params)
+#   @new_comment.author = current_user
+#   @new_comment.post = @post
+
+#   if @new_comment.save
```
Comment (anagudelogu) [inline app/controllers/comments_controller.rb]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

File: app/models/comment.rb [hunk 4]
```diff
@@ -1,9 +1,9 @@
 class Comment < ApplicationRecord
   belongs_to :author, class_name: 'User', foreign_key: 'author_id'
   belongs_to :post
-  after_create :update_comment_counter
+  # after_create :update_comment_counter
```
Comment (anagudelogu) [inline app/models/comment.rb]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

File: app/models/like.rb [hunk 5]
```diff
@@ -1,8 +1,8 @@
 class Like < ApplicationRecord
   belongs_to :author, class_name: 'User', foreign_key: 'author_id'
   belongs_to :post
-  after_create :update_like_counter
+  # after_create :update_like_counter
```
Comment (anagudelogu) [inline app/models/like.rb]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

File: app/models/post.rb [hunk 6]
```diff
@@ -2,7 +2,7 @@ class Post < ApplicationRecord
   belongs_to :author, class_name: 'User', foreign_key: 'author_id'
   has_many :comments
   has_many :likes
-  after_create :update_post_counter
+  # after_create :update_post_counter
```
Comment (anagudelogu) [inline app/models/post.rb]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @osamaashraf6, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi @osamaashraf6 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

CookedDelectableDolphin-size_restricted


## Highlights
Great job so far :heavy_check_mark: 
No linter errors :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**